Add zip template documentation (#33934)
home-assistant/home-assistant.github.io: source/_docs/configuration/templating.markdown

Note that, in Home Assistant, to convert a value to a `list`, a `string`, an `int`, or a `float`, Jinja has built-in functions with names that correspond to each type.

### Iterating multiple objects

> The `zip()` function is used to iterate multiple collections in one operation without using an indexer.

Reviewer: This sounds a bit technical. Could you add a plain English example for non-programmers, explaining how/why I would want to use this?

> The `zip()` function is used to iterate multiple collections in one operation without using an indexer.

Reviewer 2: I don't think there is a much simpler way to describe what zip does, but I would simplify it by removing Indexer.

```suggestion
The `zip()` function can be used to iterate over multiple collections in one operation.
```
